src/main/java/io/weaviate/client/v1/schema/api/ClassExists.java

```diff
@@ -0,0 +1,50 @@
+package io.weaviate.client.v1.schema.api;
+
+import io.weaviate.client.Config;
+import io.weaviate.client.base.ClientResult;
+import io.weaviate.client.base.Result;
+import io.weaviate.client.base.WeaviateError;
+import io.weaviate.client.base.WeaviateErrorMessage;
+import io.weaviate.client.base.WeaviateErrorResponse;
+import io.weaviate.client.base.http.HttpClient;
+import io.weaviate.client.v1.schema.model.WeaviateClass;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ClassExists implements ClientResult<Boolean> {
+  private final ClassGetter classGetter;
+  private String className;
+
+  public ClassExists(HttpClient httpClient, Config config) {
+    this.classGetter = new ClassGetter(httpClient, config);
+  }
+
+  public ClassExists withClassName(String className) {
+    this.className = className;
+    return this;
+  }
+
+  private Result<Boolean> toResult(WeaviateError error) {
+    return new Result<>(error.getStatusCode(), null, WeaviateErrorResponse.builder().error(error.getMessages()).build());
+  }
+
+  @Override
+  public Result<Boolean> run() {
+    if (StringUtils.isEmpty(this.className)) {
+      WeaviateErrorMessage errorMessage = WeaviateErrorMessage.builder()
+        .message("classname cannot be empty").build();
+      WeaviateErrorResponse errors = WeaviateErrorResponse.builder()
+        .error(Stream.of(errorMessage).collect(Collectors.toList())).build();
+      return new Result<>(500, null, errors);
```

Comment:
Since this is client error, 4xx http code would be more proper. Like `422 Unprocessable Content`

Reply:
Sure thing, a 4xx will be more suitable. On the same note, should the other methods for eg. https://github.com/weaviate/java-client/blob/main/src/main/java/io/weaviate/client/v1/schema/api/ClassDeleter.java#L35 be updated with the 4xx error codes if needed ?

Thanks for reviewing !

Reply:
Eventually existing 5xx codes should be replaced with 4xx, but since they may already be in use, it might be considered braking change. We will take care of them in the future and include the fix to some major/minor release.

> Thanks for reviewing !

You are welcome!